### PR TITLE
Implement tasklauncher-function

### PR DESCRIPTION
**Pull request by dturanski**

### CODE REVIEW

File: function/README.adoc [hunk 1]
```diff
@@ -18,3 +18,39 @@ Transform incoming message payload based on a SpEL expression and produces an ou
...
+}
+```
+And posted as JSON. e.g.,
+
+```json
+{"name":"timestamp"}
+```
+or
+
+[source,json]
```
Comment (artembilan) [inline function/README.adoc]: Why do you use different formats for these JSON code snippets?

I'm not sure though which one is OK or even they work both 😄

Comment (artembilan) [inline function/README.adoc]: Why do you use different formats for these JSON code snippets?

I'm not sure though which one is OK or even they work both 😄

File: function/tasklauncher-function/src/main/java/io/pivotal/java/function/tasklauncher/function/TaskLauncherFunction.java [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+
+import javax.annotation.PostConstruct;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
```
Comment (artembilan) [inline function/tasklauncher-function/src/main/java/io/pivotal/java/function/tasklauncher/function/TaskLauncherFunction.java]: May we don't use SLF4J for logging, but more generic `org.apache.commons.logging` as it is in other Spring projects?
The end-user then can decide what tool to use for the final logging.

File: function/tasklauncher-function/src/main/java/io/pivotal/java/function/tasklauncher/function/TaskLauncherFunctionProperties.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+ */
+
+package io.pivotal.java.function.tasklauncher.function;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+/**
+ * @author David Turanski
+ **/
+@ConfigurationProperties
```
Comment (artembilan) [inline function/tasklauncher-function/src/main/java/io/pivotal/java/function/tasklauncher/function/TaskLauncherFunctionProperties.java]: Why does it come without any prefix?

File: function/tasklauncher-function/src/main/java/io/pivotal/java/function/tasklauncher/function/TaskLauncherFunction.java (hunk 2 above)
Comment (dturanski) [reply function/tasklauncher-function/src/main/java/io/pivotal/java/function/tasklauncher/function/TaskLauncherFunction.java:29]: Is that true? I have seen both.

Comment (artembilan) [reply function/tasklauncher-function/src/main/java/io/pivotal/java/function/tasklauncher/function/TaskLauncherFunction.java]: A Spring Framework recommendation is to concentrate only on the `spring-jcl` usage in libraries.
But yeah... If Spring Cloud projects are based on SLF4J already, then let it be here!

**Comment by dturanski** [comment]
Merged via https://github.com/pivotal/java-functions/commit/0332b40f36afa8260acfe429e8e84c693ffebe2c

**Pull request closed without merging**